src/legacy/server/csp/index.ts

```diff
@@ -22,11 +22,7 @@ import { promisify } from 'util';
 
 const randomBytesAsync = promisify(randomBytes);
 
-export const DEFAULT_CSP_RULES = Object.freeze([
-  `script-src 'unsafe-eval' 'nonce-{nonce}'`,
```

Comment:
We can remove the `worker-src blob:` and `child-src blob:` directives because of this change.

It is unclear to me though why we need to remove the `script-src` directive as well. When this remains in the policy, mapbox-gl will not load.

Reply:
I assume we don't have any options for specifying the `nonce` attribute on the script tag which is added to the DOM?

Reply:
> I assume we don't have any options for specifying the nonce attribute on the script tag which is added to the DOM?

Correct. We don't have control over how the worker-script loaded by `mapbox-gl-csp.js`, which in turn loads content with `importScripts`. 

If we want to preserve a custom `script-src` directive with `nonce` based security (do we want this in the long run (?)), maybe we could modify how we load scripts and provided nonces on all `<script>` tags (same as now), but load `mapbox-gl-csp.js` in a separate script-tag (different as now, as it is packaged with `vendors.dll.js` now). Only for that  `mapbox-gl-csp.js`  script, we could enable `strict-dynamic`, because it's I think the only script which loads child-scripts (ie. the worker).

^ just a thought. Not 100% sure about the impact of  adding `strict-dynamic` (or the removal of `script-src` directive completely for that matter).

Reply:
> If we want to preserve a custom script-src directive with nonce based security (do we want this in the long run (?))

At this time, that's the intent. There's potential for us to switch to `script-src 'unsafe-inline' 'self'` but it's not something we have a plan to do.

>  Only for that mapbox-gl-csp.js script, we could enable strict-dynamic, because it's I think the only script which loads child-scripts (ie. the worker).

I believe that we have to add `strict-dynamic` as the CSP policy which applies to all scripts, I don't believe there's a way to scope this to only a single script tag.